build/charts/antrea/crds/externalippool.yaml

```diff
@@ -94,6 +94,93 @@ spec:
           type: date
       subresources:
         status: {}
+    - name: v1beta1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - ipRanges
+                - nodeSelector
+              properties:
+                ipRanges:
+                  type: array
+                  items:
+                    type: object
+                    oneOf:
+                      - required:
+                          - cidr
+                      - required:
+                          - start
+                          - end
+                    properties:
+                      cidr:
+                        type: string
+                        format: cidr
+                      start:
+                        type: string
+                        oneOf:
+                          - format: ipv4
+                          - format: ipv6
+                      end:
+                        type: string
+                        oneOf:
+                          - format: ipv4
+                          - format: ipv6
+                nodeSelector:
+                  type: object
+                  properties:
+                    matchExpressions:
+                      items:
+                        properties:
+                          key:
+                            type: string
+                          operator:
+                            enum:
+                              - In
+                              - NotIn
+                              - Exists
+                              - DoesNotExist
+                            type: string
+                          values:
+                            items:
+                              type: string
+                              pattern: "^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$"
+                            type: array
+                        type: object
+                      type: array
+                    matchLabels:
+                      x-kubernetes-preserve-unknown-fields: true
```

Comment:
Should we keep using  `x-kubernetes-preserve-unknown-fields: true` for `matchLabels`?

As an example, it seems kube-builder generates that schema spec for a selector:

```yaml
                         properties:
                            matchExpressions:
                              description: matchExpressions is a list of label selector
                                requirements. The requirements are ANDed.
                              items:
                                description: A label selector requirement is a selector
                                  that contains values, a key, and an operator that
                                  relates the key and values.
                                properties:
                                  key:
                                    description: key is the label key that the selector
                                      applies to.
                                    type: string
                                  operator:
                                    description: operator represents a key's relationship
                                      to a set of values. Valid operators are In,
                                      NotIn, Exists and DoesNotExist.
                                    type: string
                                  values:
                                    description: values is an array of string values.
                                      If the operator is In or NotIn, the values array
                                      must be non-empty. If the operator is Exists
                                      or DoesNotExist, the values array must be empty.
                                      This array is replaced during a strategic merge
                                      patch.
                                    items:
                                      type: string
                                    type: array
                                required:
                                - key
                                - operator
                                type: object
                              type: array
                            matchLabels:
                              additionalProperties:
                                type: string
                              description: matchLabels is a map of {key,value} pairs.
                                A single {key,value} in the matchLabels map is equivalent
                                to an element of matchExpressions, whose key field
                                is "key", the operator is "In", and the values array
                                contains only "value". The requirements are ANDed.
                              type: object
                          type: object
```

notice the use of `additionalProperties` for `matchLabels`

Please let me know what you think?

Reply:
Looks good to me. It's more specific and applies more strict validation than `x-kubernetes-preserve-unknown-fields`.

Reply:
Nothing special, just copied it from v1alpha1 definition. Do you mean that we should use `additionalProperties` instead of `x-kubernetes-preserve-unknown-fields` in `matchLabels`? Then it is more specific and strict for users to create a CR. @antoninbas @tnqn

Reply:
yes

Reply:
Thanks, got that.